src/typval.c

```diff
@@ -1578,12 +1578,15 @@ typval_compare_null(typval_T *tv1, typval_T *tv2)
 #ifdef FEAT_JOB_CHANNEL
 	    case VAR_CHANNEL: return tv->vval.v_channel == NULL;
 #endif
+	    // TODO: null_class handling
```

Comment:
Is this comment still applicable?

Reply:
Yes. I didn't do anything for or about a variable that aliases a class.  I accidentally found out that you can do
```
var vclass = C
var o = vclass.new()
```
and mentioned it in https://github.com/vim/vim/issues/12043#issuecomment-1688584460 where I questioned whether or not it was supposed to work. Other than that, I didn't try anything. I've since found in todo.txt
> When "Meta" is a class, is "const MetaAlias = Meta" allowed?  It should
> either work or given an error. possibly give an error now and implement it
> later (using a typedef).  #12006

So this is a pretty open question. Might be good idea to make the assignment fail until there's a plan on how to handle this.

I'm pretty sure it won't work in general. I don't think there is a `class` type. Certainly `var foo: class` doesn't work.

But, if you want `vclass == null` then the commented part is needed, along with plenty more.